psi4/src/psi4/libmints/basisset.cc

```diff
@@ -547,7 +547,12 @@ std::string BasisSet::print_detail_cfour() const
         for (int am = 0; am <= max_am_center; am++) {
             // Write exponents for each shell
             for (size_t ep = 0; ep < exp_per_am[am].size(); ep++) {
-                sprintf(buffer, "%14.7f", exp_per_am[am][ep]);
+                if (exp_per_am[am][ep] < 100000)
+                    sprintf(buffer, "%14.7f", exp_per_am[am][ep]);
+                else if (exp_per_am[am][ep] >= 1000000)
+                    sprintf(buffer, "%12.5f ", exp_per_am[am][ep]);
+                else if (exp_per_am[am][ep] >= 100000)
+                    sprintf(buffer, "%13.6f ", exp_per_am[am][ep]);
```

Comment:
Agreed, just ending with `else` instead of `else if (exp_per_am[am][ep] >= 100000)` will make this easier to read and will prevent potential issues if people mess with the thresholds.  Just a check, did you mean to specify different widths for each of these?  I get why different precisions would be desired, but I would expect the overall width to be the same.